nixos/modules/services/system/self-deploy.nix

```diff
@@ -67,6 +67,26 @@ in
       '';
     };
 
+    preActivationHook = lib.mkOption {
+      type = lib.types.nullOr lib.types.str;
+
+      default = null;
+
+      description = lib.mdDoc ''
+        Shell splice run before the new configuration is activated.
+      '';
+    };
+
+    postActivationHook = lib.mkOption {
+      type = lib.types.nullOr lib.types.str;
```

Comment:
Can we make these of type pkg instead? It's somewhat nicer to have an executable than a string to run.

Reply:
I want to be able to exactly match behavior on our deployed systems, and I think adding the extra indirection will make that more error prone.

Reply:
I must be missing something. I'll let you decide.